### ISSUE TITLE
[BUG] In the editor, some specific code is formatted incorrectly when using shortcuts to add comments.

### ISSUE THREAD

**Issue by F9y4ng** [violentmonkey/violentmonkey#1992]
### Problem: 

In the editor, some specific code is formatted incorrectly when using shortcuts `Ctrl + /` to add comments.

**The code is as follows:**
```JS
// ==UserScript==
// @name         test comments
// @namespace    Violentmonkey Scripts
// @match        *://example.org/*
// @grant        none
// @version      1.0
// @author       -
// @description  test comments
// ==/UserScript==

const test = "test";

const test2 =
  `<!-- test -->` +
  `<span id="${test}1">
    (${test}
    <input type="checkbox" id="${test}2"/>)
   </span>`;

console.log(test2);
```

**Gif pics for issue:**
![Pic_2023-12-31_18-16-59](https://github.com/violentmonkey/violentmonkey/assets/15855068/195f573d-cec0-4de1-b131-c0da6831046a)

![Pic_2023-12-31_18-28-14](https://github.com/violentmonkey/violentmonkey/assets/15855068/4f2caadd-46a8-466d-aad2-fdccf9c4bb72)

#### Environment:
- OS: Win10
- Browser: Chrome 120/ Firefox 121
- Violentmonkey Version: BETA 2.16.2 / 2.17.0 / BETA 2.17.1

**Comment by tophf** [comment]
Yeah looks like a bug in CodeMirror's handling of these are nested language modes.

**Comment by F9y4ng** [comment]
> Yeah looks like a bug in CodeMirror's handling of these nested language modes.

Yes, there is another, more insidious problem:

![Pic_2023-12-31_18-49-16](https://github.com/violentmonkey/violentmonkey/assets/15855068/b43890ee-cc26-4e20-b314-f51ba310a5cb)

**Comment by tophf** [comment]
Please replace gifs with actual code.

**Comment by cyfung1031** [comment]
> Yeah looks like a bug in CodeMirror's handling of these nested language modes.

not found in https://codemirror.net/ home page editor example

**Comment by tophf** [comment]
There is no example for CodeMirror 5 nested modes that also supports comment hotkey.

**Comment by cyfung1031** [comment]
> There is no example for CodeMirror 5 nested modes that also supports comment hotkey.

upgrade to CM6 or change to [Monaco Editor](https://microsoft.github.io/monaco-editor/) ?

**Comment by tophf** [comment]
No. It's not that simple, we explained it multiple times.
